pkg/vsphere/simulator/registry.go

```diff
@@ -105,3 +111,31 @@ func (r *Registry) Remove(item types.ManagedObjectReference) {
 
 	delete(r.objects, item)
 }
+
+func (r *Registry) GetEntityParent(item mo.Entity, kind string) mo.Entity {
+	for {
+		parent := item.Entity().Parent
+
+		item = Map.Get(*parent).(mo.Entity)
+
+		if item.Reference().Type == kind {
+			return item
```

Comment:
So we are guaranteed that this loop will exit on this line at some point? There is no risk the loop will iterate indefinitely?